core/src/main/java/org/elasticsearch/cluster/metadata/IndexMetaData.java

```diff
@@ -500,6 +518,10 @@ public boolean equals(Object o) {
         if (!customs.equals(that.customs)) {
             return false;
         }
+
+        if (Arrays.equals(primaryTerms, that.primaryTerms) == false) {
```

Comment:
The `version` field from this change and the `customs` field from a previous change are not incorporated into `IndexMetaData#hashCode`.